core/lib/config/src/configs/database.rs

```diff
@@ -143,12 +140,59 @@ impl DBConfig {
         60_000
     }
 
+    pub fn backup_interval(&self) -> Duration {
+        Duration::from_millis(self.backup_interval_ms)
+    }
+}
+
+/// Collection of different database URLs and general PostgreSQL options.
+/// All the entries are optional, since some components may only require a subset of them,
+/// and any component may have overrides.
+#[derive(Debug, Clone)]
+pub struct PostgresConfig {
+    /// URL for the main (sequencer) database.
+    pub master_url: Option<String>,
+    /// URL for the replica database.
+    pub replica_url: Option<String>,
+    /// URL for the prover database.
+    pub prover_url: Option<String>,
+    /// Maximum size of the connection pool.
+    pub max_connections: Option<u32>,
+    /// Statement timeout in seconds for Postgres connections. Applies only to the replica
+    /// connection pool used by the API servers.
+    pub statement_timeout_sec: Option<u64>,
+}
+
+impl PostgresConfig {
+    /// Returns a copy of the master database URL as a `Result` to simplify error propagation.
+    pub fn master_url(&self) -> anyhow::Result<String> {
+        self.master_url
+            .clone()
+            .ok_or_else(|| anyhow::anyhow!("Master DB URL is absent"))
```

Comment:
Nit: You can use the `anyhow::Context` trait to convert `Option`s with a similar result.